# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.cpp
# Commit: b6d68da027a2
# Full Hash: b6d68da027a23da0a86fd73efc2fc004a2e763ba
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2023-07-04 21:49:05
# Regressor Bug: 1837620
# File Overlap Count: 1
# Description:
#   Bug 1837620 - Part 6: Make edges for multiple shape guard weak too r=sfink
#   
#   To make the edges for the multiple shape guard weak we need a way to find and
#   update the objects used to hold these edges. The simplest way to do this is to
#   add a vector in the zone.  I don't believe that there are not huge numbers of
# ==============================================================================

diff -r 318af9553464 -r b6d68da027a2 js/src/jit/MacroAssembler.cpp
--- a/js/src/jit/MacroAssembler.cpp	Tue Jul 04 13:58:57 2023 +0000
+++ b/js/src/jit/MacroAssembler.cpp	Tue Jul 04 13:58:57 2023 +0000
@@ -4454,14 +4454,9 @@
   Label* onMatch = cond == Assembler::Equal ? label : &done;
 
   // Load the object's shape pointer into shapeScratch, and prepare to compare
-  // it with the shapes in the list. On 64-bit, we box the shape. On 32-bit,
-  // we only have to compare the 32-bit payload.
-#ifdef JS_PUNBOX64
-  loadPtr(Address(obj, JSObject::offsetOfShape()), endScratch);
-  tagValue(JSVAL_TYPE_PRIVATE_GCTHING, endScratch, ValueOperand(shapeScratch));
-#else
+  // it with the shapes in the list. The shapes are stored as private values so
+  // we can compare directly.
   loadPtr(Address(obj, JSObject::offsetOfShape()), shapeScratch);
-#endif
 
   // Compute end pointer.
   Address lengthAddr(shapeElements,
